src/workers/parser/getSymbols.js

```diff
@@ -123,6 +124,57 @@ function getFunctionParameterNames(path: SimplePath): string[] {
   return [];
 }
 
+function getFunctionSignature(
+  path: SimplePath | null,
+  signature: string | null
+) {
+  if (!path) {
+    return signature;
+  }
+
+  if (t.isFunctionDeclaration(path)) {
+    if (path.node.id == null) {
+      return "anonymous";
+    }
+    return path.node.id.name;
+  }
+
+  if (!signature) {
+    signature = getFunctionName(path.node, path.parent);
+    path = path.parentPath && path.parentPath.parentPath;
+  }
+
+  if (t.isMemberExpression(path)) {
+    if (!path || !path.node.object.name) {
```

Comment:
I think duck typing like this makes the code harder to follow. I'd prefer `!path || !t.isIdentifier(path.node.object)`.

Reply:
nice didn't think about that .. thansk!